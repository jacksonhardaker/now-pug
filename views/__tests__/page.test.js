import pug from "pug";
import path from "path";
import {
  getByLabelText,
  getByText,
  getByTestId,
  queryByTestId,
  // Tip: all queries are also exposed on an object
  // called "queries" which you could import here as well
  wait,
} from "@testing-library/dom"
import "@testing-library/jest-dom/extend-expect"

const renderPug = data => pug.renderFile(path.join(__dirname, "..", "page.pug"), data);

test("It should render pug correctly", () => {
  expect(renderPug({
    title: "Pug",
    text: "Pug is great"
  })).toMatchSnapshot();
});

test("Heading should be in the DOM", () => {
  const div = document.createElement("div");
  div.innerHTML = renderPug({
    title: "Pug",
    text: "Pug is great"
  });

  getByTestId(div, "page-heading");
});
