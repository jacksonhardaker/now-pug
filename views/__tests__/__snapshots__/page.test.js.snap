// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render pug correctly 1`] = `"<!DOCTYPE html><html lang=\\"en-US\\"><head><title>Pug</title><meta charset=\\"utf-8\\"><meta name=\\"description\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"></head><body><h1>Pug</h1><p>Pug is great</p></body></html>"`;
